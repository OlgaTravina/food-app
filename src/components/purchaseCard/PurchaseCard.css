.purchaseCard__card {
	border-radius: 19px;
	padding:10px;
	border: 1.57px solid rgba(237, 242, 247, 1);
	margin-bottom: 20px;
}

.purchaseCard__title {
	font-family: Manrope;
	font-size: 19px;
	font-weight: 800;
	line-height: 28px;
	letter-spacing: 0.6px;
	text-align: left;
	color: rgba(26, 32, 44, 1);
}

.purchaseCard__text {
    font-family: Manrope;
	font-size: 16px;
	font-weight: 500;
	line-height: 24px;
	letter-spacing: 0.6px;
	text-align: left;
	color: rgba(160, 174, 192, 1);
}

.purchaseCard__price {
    font-family: Manrope;
	font-size: 22px;
	font-weight: 800;
	line-height: 33px;
	letter-spacing: 0.3;
	text-align: right;
	color: rgba(26, 32, 44, 1);
}