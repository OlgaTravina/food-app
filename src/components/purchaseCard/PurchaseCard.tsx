import './PurchaseCard.css';

type PurchaseCardProps = { 
	imgUrl: string,	
	title: string,
	text: string,
	price: string,
	urlPurchas: string
 }

 const innerStyle = {
    display:"flex",
	gap:"20px",
	alignItems:"center"		
 }


function PurchaseCard({imgUrl,title,text,price,urlPurchas}:PurchaseCardProps) {
 
	return (
		<div className='purchaseCard__card'>
			<div style={innerStyle}>
				<img className="" src={imgUrl} alt="" />		
			    <div>
				  <div className='purchaseCard__title'>{title}</div>
			      <div className='purchaseCard__text'>{text}</div> 			
			    </div>
				<span className='purchaseCard__price' style={{marginLeft:"auto"}}>{price}</span>	
			</div>			
			<img style={{minWidth:"100%"}}  src={urlPurchas} alt="" />
		</div>
	)
}
export default PurchaseCard;