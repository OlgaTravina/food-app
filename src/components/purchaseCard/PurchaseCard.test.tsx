import { render} from "@testing-library/react";
import PurchaseCard from "./PurchaseCard";

test("Snapshot PurchaseCard", () => {
  const testData = {
    imgUrl: "/images/Control Purchase/icon (1).svg",
    title: "Expense",
    text: "Increased By 10%",
    price: "$409.00",
    urlPurchas: "/images/Control Purchase/divider (1).svg",
  };

  const {container} = render(<PurchaseCard {...testData} />);

  expect(container).toMatchSnapshot();
});

