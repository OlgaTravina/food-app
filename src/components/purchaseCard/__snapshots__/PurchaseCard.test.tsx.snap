// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot PurchaseCard 1`] = `
<div>
  <div
    class="purchaseCard__card"
  >
    <div
      style="display: flex; gap: 20px; align-items: center;"
    >
      <img
        alt=""
        class=""
        src="/images/Control Purchase/icon (1).svg"
      />
      <div>
        <div
          class="purchaseCard__title"
        >
          Expense
        </div>
        <div
          class="purchaseCard__text"
        >
          Increased By 10%
        </div>
      </div>
      <span
        class="purchaseCard__price"
        style="margin-left: auto;"
      >
        $409.00
      </span>
    </div>
    <img
      alt=""
      src="/images/Control Purchase/divider (1).svg"
      style="min-width: 100%;"
    />
  </div>
</div>
`;
